### PR TITLE
Annotation rendering refactor

### LINKED ISSUE #6: Loupe annotation artifacts
When using the loupe annotation, I get the following [artifact](https://user-images.githubusercontent.com/987398/29037490-e7213d02-7b71-11e7-861f-c0ea99f4f07b.png). This artifact only occurs occasionally. Other times, the button simply doesn't get rendered at all. I tested this on an Android device with API 25.

### PR DESCRIPTION
**Pull request by alexfu**
Fixes #6.

I actually had not intended for this PR to fix issue #6 but it just kind of turned out that way. This PR was meant to refactor the annotation rendering code such that it could be easily isolated from the `ScreenshotAnnotatorActivity`. Doing so would have made debugging #6 easier. In any case, here is a summary of the changes made:

- Moved annotation rendering into a new `AnnotationView`
  - This view is responsible for taking an `Annotation` and rendering it on the screen in the correct order.
- `Annotation`s are now owners of `Renderer`s. 
- `ArrowAnnotationView`, `LoupeAnnotationView`, and `BlurAnnotationView` have been superseded by the new `AnnotationView`.
- Smoother borders around loupe annotations.

There's still some final things I'd like to clean up so in the meantime, this can be tested and reviewed.

### CODE REVIEW

File: src/main/java/com/buglife/sdk/AnnotationView.java [hunk 1]
```diff
@@ -1,95 +1,322 @@
-/*
```
Comment (schukin) [inline src/main/java/com/buglife/sdk/AnnotationView.java]: Whoops! Can we leave this in? 😇

**Review by schukin — CHANGES_REQUESTED** [review]
This is *awesome*! Just two comments:

* Looks like the license may have been accidentally removed from AnnotationView.java
* Tried running on a device, and it appears that the resulting image is a bit pixelated, almost as though it's not respecting the screen density... This is particularly notable when drawing an annotation, saving, then opening the same screenshot again (the previously drawn annotation is noticeably pixelated). I thought this may perhaps have something to do with canvas.drawBitmap() and the matrix being applied, but that's of course necessary to scale the image. Any ideas?

**Comment by alexfu** [comment]
@schukin Updated. Apparently the pixelation was happening because the bitmap was being drawn with nearest neighbor sampling (not entirely sure what sampling does `¯\_(ツ)_/¯`). Enabling `FILTER_BITMAP_FLAG` changes this to bilinear sampling which results in a smoother image.

https://developer.android.com/reference/android/graphics/Paint.html#FILTER_BITMAP_FLAG

**Comment by schukin** [comment]
Nice! Looks great.

One more regression I just noticed: Drawing a 2nd annotation (i.e. two consecutive arrows) removes the first annotation. Once this gets fixes I'll be happy to merge! (Sorry, looks like it's on us to add some automated tests soon 😬)

**Comment by alexfu** [comment]
@schukin alright, should be good now!

**Comment by schukin** [comment]
Awesome!!

I loathe to say this, but I did notice one more issue when testing out the latest revisions... Drawing / mutating annotations is *really* slow in terms of FPS. This seems to be caused by the screenshot being continuously redrawn.

That said, I'm going to go ahead and merge this because (1) these changes are a huge step in the right direction (2) it pains me to see large change sets stuck in review hell forever (3) we don't yet have any contributor guidelines :)

I've created a separate issue (#10) to track the FPS drop. Once again, awesome work 🤘

**Pull request merged**